start_control_ssm.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -e
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register with a hybrid activation
+  activation_code=$(fetch_from_json '.["ssm"]["activation-code"]' "${USER_DATA}")
+  activation_id=$(fetch_from_json '.["ssm"]["activation-id"]' "${USER_DATA}")
+  region=$(fetch_from_json '.["ssm"]["region"]' "${USER_DATA}")
+
+  # Register with AWS Systems Manager (SSM)
+  if ! amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"; then
+
+    # Print errors line-by-line from ssm agent error log,
+    # as they don't print to the EC2 console otherwise.
+    while IFS= read -r line; do
+      log "${line}"
+    done < "/var/log/amazon/ssm/errors.log"
```

Comment:
Does printing the log file directly to the console (without chunking lines) cause the output to become garbled?

I'd prefer to see this written with a simpler, single command rather than iteration:

```bash
cat /var/log/amazon/ssm/errors.log >&2
```

In fact, as is (and with the above) the log file would grow after each attempt and keep on going.. so really, we should truncate the file when attempting to register. To keep around some of the older logs, we can do a cheap "logrotate" and rename the existing files to allow for debugging while allowing SSM to write its errors to the fresh log. This would ensure that there wasn't **too much printed** to the console (making `cat` a practical choice) and that the size of the log doesn't become overwhelming over time.

Reply:
I can't say I've seen it garbled personally, so it may not be necessary. I'm definitely keen on using a simple `cat` if nobody has concerns.

Regarding truncating the file, from what I saw in my testing the errors.log did not survive between attempts, most likely since it's not part of the persistent storage. That said, if folks would feel more comfortable with a logrotate I can certainly add it.